src/Compilers/CSharp/Portable/Emitter/Model/NamedTypeSymbolAdapter.cs

```diff
@@ -382,9 +382,18 @@ Cci.ITypeReference Cci.ITypeDefinition.GetBaseClass(EmitContext context)
 
             if (AdaptedNamedTypeSymbol is SourceMemberContainerTypeSymbol container)
             {
+                var interfaces = container.GetInterfacesToEmit();
+
                 foreach ((MethodSymbol body, MethodSymbol implemented) in container.GetSynthesizedExplicitImplementations(cancellationToken: default).MethodImpls)
                 {
                     Debug.Assert(body.ContainingType == (object)container);
+
+                    // Avoid emitting duplicate methodimpl entries (e.g., when the class implements the same interface twice with different nullability).
```

Comment:
I didn't understand which test was actually exercising this code path.

Reply:
Tests 01, 02, 03 hit this. It's about synthesized methodimpl entries.
Test 04 verifies pre-existing behavior.
Test 05 exercises the changes in MethodCompiler.cs and SynthesizedMetadataCompiler.cs (depending on the `bool metadataOnly` theory parameter). That's about synthesized forwarding methods.